udata/harvest/tests/test_base_backend.py

```diff
@@ -281,6 +287,78 @@ def test_harvest_datasets_get_deleted(self):
         for item in job.items:
             assert item.dataset is None
 
+    def test_no_datasets_duplication(self, app):
+        limit = app.config["HARVEST_AUTOARCHIVE_GRACE_DAYS"]
+        last_update = datetime.utcnow() - timedelta(days=limit - 1)
+        reuse_remote_id_uri = "http://example.com/reuse-this-uri"
+        nb_datasets = 3
+        source = HarvestSourceFactory(
+            config={
+                "nb_datasets": nb_datasets,
+                "dataset_remote_ids": [reuse_remote_id_uri],
+            }
+        )
+        backend = FakeBackend(source)
+
+        # Create a dataset that should be reused by the harvest, which will update it
+        # instead of creating a new one, as it has the same remote_id, domain and source_id.
+        dataset_reused = DatasetFactory(
+            title="Reused Dataset",
+            harvest={
+                "domain": source.domain,
+                "remote_id": "fake-0",  # the FakeBackend harvest should reuse this dataset
+                "source_id": str(source.id),
+                "last_update": last_update,
+            },
+        )
+
+        # Create a dataset that should be reused even though it's a different `remote_id` and `domain,
+        # because it's an URI.
```

Comment:
Did you mean?
```suggestion
        # Create a dataset that should be reused even though it's a different `source_id` and `domain,
        # because the remote_id is the same and an URI.
```

Reply:
Yeah, it's wayyyyy clearer that way, thanks 😅